AuthenticatorShared/Core/Vault/Repositories/AuthenticatorItemRepository.swift

```diff
@@ -176,14 +197,35 @@ extension DefaultAuthenticatorItemRepository: AuthenticatorItemRepository {
         return try await items.asyncMap { item in
             try await cryptographyService.decrypt(item)
         }
-        .compactMap { $0 }
     }
 
     func fetchAuthenticatorItem(withId id: String) async throws -> AuthenticatorItemView? {
         guard let item = try await authenticatorItemService.fetchAuthenticatorItem(withId: id) else { return nil }
         return try? await cryptographyService.decrypt(item)
     }
 
+    func refreshTotpCodes(on items: [ItemListItem]) async throws -> [ItemListItem] {
+        try await items.asyncMap { item in
+            guard case let .totp(model) = item.itemType,
+                  let key = model.itemView.totpKey,
+                  let keyModel = TOTPKeyModel(authenticatorKey: key)
+            else {
+                errorReporter.log(error: TOTPServiceError
+                    .unableToGenerateCode("Unable to refresh TOTP code for list view item: \(item.id)"))
+                return item
```

Comment:
Can we test this?

Reply:
Not as far as I can tell, unfortunately. In practice I don't think we'll ever actually get here, because we don't have a `itemType` other than `.totp` at the moment, and if the type *is* TOTP an `ItemListItem` can't be created unless the key is valid. However, because the contained `AuthenticatorItemView` has an optional `totpKey` and `TOTPKeyModel.init()` is a failable initializer, we still have to have code to handle if they fail, even though in practice they won't.